### PR TITLE
wayland: bump expat

### PR DESCRIPTION
**Pull request by ericLemanissier**
Specify library name and version:  **wayland/***

- [X] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [X] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [ ] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by ericLemanissier** [comment]
I'll add components later today

**Review by SSE4 — DISMISSED** [review]

**Review by prince-chrismc — APPROVED** [review]

**Comment by ericLemanissier** [comment]
this blocks #3663

**Comment by prince-chrismc** [comment]
tag the reviewers #2857

File: recipes/wayland/all/conanfile.py [hunk 1]
```diff
@@ -71,8 +77,31 @@ def package(self):
...
     def package_info(self):
-        self.cpp_info.libs = tools.collect_libs(self)
+        self.cpp_info.components["wayland-scanner"].names["pkg_config"] = "wayland-scanner"
+        self.cpp_info.components["wayland-scanner"].requires = ["expat::expat"]
+        if self.options.enable_dtd_validation:
+            self.cpp_info.components["wayland-scanner"].requires.append("libxml2::libxml2")
+
+        if self.options.enable_libraries:
+            self.cpp_info.components["wayland-server"].libs = ["wayland-server"]
+            self.cpp_info.components["wayland-server"].names["pkg_config"] = "wayland-server"
```
Comment (uilianries) [inline recipes/wayland/all/conanfile.py:89]: what about CMake targets?

Comment (ericLemanissier) [reply recipes/wayland/all/conanfile.py:89]: they don't exist AFAIK https://packages.ubuntu.com/groovy/amd64/libwayland-dev/filelist

**Review by uilianries — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]

**Pull request merged**